src/main/java/duke/command/EventCommand.java

```diff
@@ -0,0 +1,75 @@
+package duke.command;
+
+import duke.exception.DukeException;
+import duke.storage.Storage;
+import duke.task.Event;
+import duke.task.TaskList;
+import duke.ui.Ui;
+
+/**
+ * A subclass of Command that represents the command
+ * to add an event task into the TaskList.
+ * @author Oskar Lew
+ * @version 0.1
+ * @since 0.1
+ */
+public class EventCommand extends Command {
+
+    /**
+     * Constructor of EventCommand.
+     * @param command Command from the user.
+     */
+    public EventCommand(String[] command) {
+        super(command);
+    }
+
+    /**
+     * Method to add an event task to the TaskList.
+     * @param tasks List of tasks.
+     * @param ui Ui of the chat.
+     * @param storage Storage of Duke.
+     * @return The confirmation message.
+     */
+    public String execute(TaskList tasks, Ui ui, Storage storage) {
+        try {
+            StringBuilder stringBuilder = new StringBuilder();
+            StringBuilder startTime = new StringBuilder();
+            StringBuilder endTime = new StringBuilder();
+            boolean hasFrom = false;
+            boolean hasTo = false;
+            for (int i = 1; i < command.length; i++) {
```

Comment:
Is there a more efficient way to do this? Feels Like a rather long and complicated method, just to segment the "/to" and "/from" strings. I understand you are working with an array, but perhaps rejoining the command string form [1:end] and using the String.split() method could make your life easier?